regtest.py

```diff
@@ -1293,5 +1296,17 @@ def email_developers():
 
 
 if __name__ == "__main__":
-    n = test_suite(sys.argv[1:])
-    sys.exit(n)
+    temp_dir = tempfile.gettempdir()
+    temp_file = os.path.join(temp_dir, 'amrex_regtest_lock_file_'+getpass.getuser())
+    test_lock = fasteners.InterProcessLock(temp_file)
+    gotten = test_lock.acquire(blocking=False)
+    status = 255
+    try:
+        if gotten:
+            status = test_suite(sys.argv[1:])
+        else:
+            print("Wait please! Another process is running regression tests.")
+    finally:
```

Comment:
What would happen if we never reach finally, e.g., if the `try:` block calls `sys.exit` or if some other reason aborts mission. Should we somehow clean up the file lock on fresh runs?

I am also a bit confused with the `tempdir` above: would multiple users not receive different temp dirs and thus never see the same locks?

Reply:
If a process dies without calling `release`, a fresh run will still be able to acquire the lock, because the lock is on the number of file handles not on the existence of the file.

This is meant to limit multiple runs by the same user, not among different users.  At CCSE, we use a shared account on our regression testing machine.  Multiple people use that account to run jobs manually.  People often started a run without realizing another person had already started a run.  The second run would kill the first one making the first person wonder what had just happened.